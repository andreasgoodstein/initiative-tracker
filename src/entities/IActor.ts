export default interface IActor {
    readonly name: string;
    readonly initiativeBonus: number;

    initiativeTotal?: number;
    initiativeRoll?: number;
}
