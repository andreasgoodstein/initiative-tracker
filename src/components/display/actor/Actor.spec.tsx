import { shallow } from 'enzyme';
import React from 'react';

import IActor from 'entities/IActor';

import Actor from './index';

const testActor: IActor = {
    initiativeBonus: 30,
    name: 'Test Actor',
};

describe('the Actor component', () => {
    it('renders', () => {
        const wrapper = shallow(<Actor actor={testActor}/>);

        expect(wrapper).not.toBeNull();
        expect(wrapper).toMatchSnapshot();
    });
});
