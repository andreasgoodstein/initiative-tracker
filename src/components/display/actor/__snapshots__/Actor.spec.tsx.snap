// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Actor component renders 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "borderColor": "#fff",
      "borderWidth": 5,
      "flexDirection": "row",
      "height": 70,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <Component
    style={
      Object {
        "fontSize": 18,
        "paddingRight": 10,
      }
    }
  >
    Test Actor
  </Component>
  <Component
    style={
      Object {
        "fontSize": 18,
        "paddingRight": 10,
      }
    }
  >
    30
  </Component>
</Component>
`;
